### PR TITLE
sync with master, bump version to 2.2.0, style fixes, enhance tests

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

File: lib/vsc/utils/generaloption.py [hunk 1]
```diff
@@ -682,12 +688,12 @@ def get_env_options(self):
...
         if candidates:
-            msg="Found %s environment variable(s) that are similar but do not match valid option(s): %s"
+            msg = "Found %s environment variable(s) that are prefixed with %s but do not match valid option(s): %s"
+            tup = (len(candidates), self.envvar_prefix, ','.join(candidates))
             if self.error_env_options:
-                func=self.log.error
+                self.error_env_option_method(msg, *tup)
             else:
-                func=self.log.debug
-            func(msg, len(candidates), ",".join(candidates))
```
Comment (stdweird) [inline lib/vsc/utils/generaloption.py]: what's wrong with the `func` ? at least you don't need one of those `tup`. and leave the default to `log.debug`.

Comment (boegel) [inline lib/vsc/utils/generaloption.py]: well, neither of them are a function, both are methods ;)

I'll restore your approach and go back to debug.

**Comment by boegel** [comment]
@stdweird: fixed your remark

**Comment by stdweird** [comment]
@boegel merged, thanks for the PR!

**Pull request merged**